hostsfile/src/lib.rs

```diff
@@ -206,9 +210,39 @@ impl HostsBuilder {
         let begin = lines.iter().position(|line| line.trim() == begin_marker);
         let end = lines.iter().position(|line| line.trim() == end_marker);
 
+        let mut lines_to_insert = vec![];
+        if !self.hostname_map.is_empty() {
+            writeln!(lines_to_insert, "{begin_marker}")?;
+            for (ip, hostnames) in &self.hostname_map {
+                if cfg!(windows) {
+                    // windows only allows one hostname per line
+                    for hostname in hostnames {
+                        writeln!(lines_to_insert, "{ip} {hostname}")?;
+                    }
+                } else {
+                    // assume the same format as Unix
+                    writeln!(lines_to_insert, "{} {}", ip, hostnames.join(" "))?;
+                }
+            }
+            writeln!(lines_to_insert, "{end_marker}")?;
+        } else {
+            // There's no need to write to the hosts file if the map is empty
+            return Ok(false);
+        }
```

Comment:
I think there is one corner case: when the map _becomes_ empty. Maybe this cannot happen in innernet as the current peer is probably always online, but `hostsfile` is a semi-independent crate, so it would be better not to have this assumption.

The old code still wrote the file but skipped the begin and end markers, which is I think a good behavior.

Reply:
`else` removed :heavy_check_mark: